sdk/eventhub/event-hubs/test/internal/node/disconnect.spec.ts

```diff
@@ -123,10 +123,12 @@ testWithServiceTypes((serviceVersion) => {
           await receiver1.connect({
```

Comment:
how is this `prefetch` count support related to the user issue?
I thought the user issue was about asking for a minimum number of messages?
Also, I don't think this was a feature in 5.8.0, how was rolling back to 5.8.0 was fixing the issue for the user?

Reply:
It is not related. This PR improves performance by prefetching events so chances are the received batch size will be larger than before. We still don't want to wait until we receive `maxBatchSize` because that is against the library goal of maximizing throughput. 

Now there is a question for you, how can we analyze the performance characteristic of this change? More specifically, can we show the average/median batch size before and after the change? can we show how often the user callback was called per minute or second?

Reply:
Yeah, these are good metrics to track, how often the user callback is called is already being tracked in the perf test.
We might need to update the framework/test to track the batch size.

Let's work on testing this, so we don't miss anything obvious. 🙂

Reply:
Here is a stress test run for the checkpoint store test..
`Red ❤️` is the code from this PR, and `Blue 💙` is the 5.10.0 version.

![image](https://github.com/Azure/azure-sdk-for-js/assets/10452642/8eb7e183-2639-42f2-88fc-2bafd1f59a83)

@deyaaeldeen and I were discussing... 
and we speculate the bigger spikes in the CPU usage to be when the credits are added for the new events to be received, with the new pre-fetch method, the spikes are lot smaller and less dense.
The memory usage graph also validates the prefetch's improvement.

Reply:
Thanks a lot @HarshaNalluru!

Also, the [performance analysis](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2824195&view=logs&j=42ded549-05ee-5cdb-7ba7-7a948a0cc056&t=49e32a0b-4245-531b-0723-9d45113ffbd3) shows that the size median of the received batches increased as well as max and mean throughput.